docs/api_requests/block-vhost-user.md

```diff
@@ -0,0 +1,201 @@
+# Vhost-user block device
+
+As an alternative to [file-backed block device](block-io-engine.md) `Sync` and
+`Async` engines, Firecracker supports a vhost-user block device.
+
+There is a good introduction of how a vhost-user block device works
+in general at [FOSDEM23](https://archive.fosdem.org/2023/schedule/event/sds_vhost_user_blk).
+
+[Vhost-user](https://qemu-project.gitlab.io/qemu/interop/vhost-user.html)
+is a userspace protocol that allows to delegate Virtio queue processing
+to another userspace process on the host, as opposed to performing this task
+within Firecracker's VMM thread.
+
+In the vhost-user architecture, the VMM acts as a vhost-user frontend and
+it is responsible for:
+
+- connecting to the backend via a Unix domain socket (UDS)
+- feature negotiation with the backend and the guest
+- handling device configuration requests from the guest
+- sharing sufficient information about the guest memory and Virtio queues
+  with the backend
+
+The vhost-user backend receives the information from the frontend and performs
+handling of IO requests from the guest.
+
+The UDS socket is only used for control plane purposes
+and does not participate in the data plane.
+
+Firecracker only implements a vhost-user frontend. Users are free
+to choose from [existing open source backends](#backends) or implement their
+own.
+
+## Topology
+
+Each vhost-user device connects to its own UDS socket. There is no way for
+multiple devices to share a single socket, as there is no way to differentiate
+messages related to devices at the vhost-user protocol level.
+
+Each device can be served by a separate backend or a single backend can serve
+multiple devices.
+
+## Interactions with the backend
+
+There are two points when the vhost-user frontend communicates with
+the backend:
+
```

Comment:
I saw you mention PATCH in the `patch-block.md`. In this case there are 3 places where FC communicates with backend.

Reply:
That is a good point. Updated.